### PR TITLE
Feature: compassFadeWhenNorth

### PR DESCRIPTION
**Pull request by mysport12**
## Description

Added the ability to control whether or not the compass fades out when pointing north. On Android the default was true, and on iOS the default was false. This brings consistency across platforms by defaulting to true ("breaking" for iOS) and adds the ability to toggle this behavior on/off with a new prop

## Checklist

- [x] I have tested this on a device/simulator for each compatible OS
- [x] I updated the documentation with running `yarn generate` in the root folder
- [ ] I mentioned this change in `CHANGELOG.md`
- [x] I updated the typings files (`index.d.ts`)
- [ ] I added/ updated a sample (`/example`)

### CODE REVIEW

**Review by mfazekas — APPROVED** [review]

File: ios/RCTMGL-v10/RCTMGLMapView.swift [hunk 1]
```diff
@@ -155,7 +155,11 @@ open class RCTMGLMapView : MapView {
   }
   
   @objc func setReactCompassEnabled(_ value: Bool) {
-    mapView.ornaments.options.compass.visibility = value ? .visible : .hidden
+    mapView.ornaments.options.compass.visibility = value ? .adaptive : .hidden
+  }
+  
+  @objc func setReactCompassFadeWhenNorth(_ value: Bool) {
+    mapView.ornaments.options.compass.visibility = value ? .adaptive : .visible
```
Comment (mfazekas) [inline ios/RCTMGL-v10/RCTMGLMapView.swift:162]: I'm not sure if this logic works with `compassEnabled={false} compassFadeWhenNorth={false}`, I think it just will show the compass

**Review by mfazekas — CHANGES_REQUESTED** [review]
Thanks much looks awesome. 

I think android logic might need an update.

I'm also thinking about setting the default it to `false`. 
It could be weird if someone set just `compassEnable={true}` and then doesn't sees the compass because it's hidden as it's north.

Whay do you think?!

File: ios/RCTMGL-v10/RCTMGLMapView.swift (hunk 1 above)
Comment (mysport12) [reply ios/RCTMGL-v10/RCTMGLMapView.swift:162]: Good catch and valid point! I will default to false, and add some checks to make sure it works in all scenarios

Comment (mysport12) [reply ios/RCTMGL-v10/RCTMGLMapView.swift]: I pushed an update. Tested on iOS for the logic issue you caught above. I also noticed that Android/iOS had different defaults for compassEnabled. I set both platforms to false for consistency but feel free to change them to true by default.

**Comment by mfazekas** [comment]
@mysport12 generated should commited according to CI: https://github.com/rnmapbox/maps/runs/7488241923?check_suite_focus=true

**Comment by mysport12** [comment]
Should be all set now

**Review by mfazekas — APPROVED** [review]

**Pull request merged**